### DISCUSSION TITLE
vlm启动，识别效果有问题

### DISCUSSION THREAD

**Discussion by LeeLear** [magicpdf/Magic-PDF#4323] [Q&A]
1、物理机器
昇腾910b3

2、镜像
export IMAGE=quay.io/ascend/vllm-ascend:v0.13.0rc1
docker run -itd --shm-size 500g \
    --privileged \
    --name vllm-ascend-mineru13 \
    --device /dev/davinci6 \
    --device /dev/davinci7 \
    --device /dev/davinci_manager \
    --device /dev/devmm_svm \
    --device /dev/hisi_hdc \
    -e ASCEND_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
    -v /usr/local/dcmi:/usr/local/dcmi \
    -v /usr/local/bin/npu-smi:/usr/local/bin/npu-smi \
    -v /usr/local/Ascend/driver/lib64/:/usr/local/Ascend/driver/lib64/ \
    -v /usr/local/Ascend/driver/version.info:/usr/local/Ascend/driver/version.info \
    -v /etc/ascend_install.info:/etc/ascend_install.info \
    -v /root/.cache:/root/.cache \
    -v /etc/hccn.conf:/etc/hccn.conf \
    -v /iflytek:/iflytek \
    -p 9005:9005 \
    -p 30000:30000 \
    $IMAGE /bin/bash

3、进入镜像后环境安装
pip install -U 'mineru[core]' -i https://mirrors.aliyun.com/pypi/simple
mineru-models-download -s modelscope -m all

Centos系统: 
yum install -y mesa-libGL mesa-libGL-devel libXrender libSM libXext tesseract tesseract-langpack-chi_sim
Ubuntu系统:
apt-get update
# OpenGL & X11 库
apt-get install -y libgl1-mesa-glx libgl1-mesa-dev \
                   libxrender1 libsm6 libxext6
# OCR & 中文语言包
apt-get install -y tesseract-ocr tesseract-ocr-chi-sim

pip install numpy==1.26.4 -i https://mirrors.aliyun.com/pypi/simple

4、启动服务

export MINERU_MODEL_SOURCE=local  
mineru-vllm-server --port 30000

5、调用解析
一张图片大部分是空白，少部分是文字
<img width="812" height="750" alt="image" src="https://github.com/user-attachments/assets/55de22e6-cf12-486a-ad3e-88f0cc770359" />

6、返回结果
VLM 模型回复：
 095 097 238 110header
794 97 908 110header
521 185 879 217title
96 223 899 278text
96 280 899 300text
96 302 899 322text
96 324 899 344text
96 346 899 366text
96 368 899 388text
96 390 899 409text
96 411 899 432text
96 434 899 454text
96 456 899 476text
96 478 899 498text
96 500 899 519text
96 521 899 542text
96 544 899 564text
96 566 899 586text
96 588 899 608text
96 610 899 630text
96 632 899 653text
96 655 899 675text
96 677 899 697text
96 700 899 719text
96 721 899 742text
96 744 899 763text
96 765 899 784text
96 787 899 807text
96 809 899 829text
96 831 899 850text
96 852 899 872text
96 874 899 894text
96 896 899 916text
96 918 899 938text
96 940 899 959text
96 961 899 980text
96 982 899 999text

可以分析下原因吗

**Comment by dosubot** [comment]
Hi @LeeLear! I'm [Dosu](https://go.dosu.dev/dosubot) and I’m helping the MinerU team.


识别效果不理想，主要原因有以下几个：

1. **VLM后端在昇腾NPU上功能受限**：MinerU的VLM模式（如vllm-engine）对昇腾910B3等NPU没有专门适配，部分关键特性（如自定义logits processor）只在NVIDIA CUDA显卡上启用，NPU上会被禁用，导致识别结果和官方demo（通常用NVIDIA GPU）有明显差异，甚至输出为空或只有框无内容。[相关讨论](https://github.com/opendatalab/MinerU/issues/3682)

2. **默认不做图片内文字OCR**：VLM模型默认只识别图片整体和caption，不会自动对图片里的嵌入文字做OCR。如果需要识别图片里的文字，必须在命令行加 `--method ocr` 或在配置中显式开启OCR，否则只会输出框和类型（如header、title、text），不会有具体文字内容。[相关说明](https://github.com/opendatalab/MinerU/discussions/3728)

3. **复杂布局/低分辨率图片易识别失败**：图片分辨率低、内容复杂或大面积空白时，VLM模型容易置信度不足或误判区域为图片，导致输出只有框或原图，内容缺失。[相关问题](https://github.com/opendatalab/MinerU/issues/3852)

4. **依赖和模型需保持最新**：MinerU、mineru-vl-utils、vllm等依赖需与官方推荐版本一致，模型需用 `mineru-models-download` 下载最新，配置文件（如mineru.json）也要同步，否则容易初始化失败或识别效果差。[相关建议](https://github.com/opendatalab/MinerU/issues/3280)

**建议排查和优化：**
- 启动时加 `--method ocr` 或 `force_OCR`，确保图片内文字能被识别。
- 尝试用 pipeline 后端（`-b pipeline`），该模式对NPU支持更好，识别更稳定。
- 检查所有依赖和镜像版本是否最新，模型和配置是否与官方一致。
- 优化图片质量（裁剪、旋转、增强对比度、提高分辨率），复杂场景建议先预处理图片。
- 查看中间文件（如 middle.json、discordblocks），分析哪些内容被遗漏或误判。
- 如仍有问题，建议关注后续版本更新或考虑用NVIDIA GPU测试关键场景。

相关详细说明和社区反馈可参考：[VLM识别NPU适配问题](https://github.com/opendatalab/MinerU/issues/3682)、[VLM默认不OCR说明](https://github.com/opendatalab/MinerU/discussions/3728)、[复杂图片识别bug](https://github.com/opendatalab/MinerU/issues/3852)。



*To reply, just mention [@dosu](https://go.dosu.dev/dosubot?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=reply-with-mention&utm_term=opendatalab%2FMinerU).*

---

Share context across your team and agents. Try [Dosu](https://dosu.dev?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=knowledge-infrastructure-tagline&utm_term=opendatalab%2FMinerU).

[![Leave Feedback](https://img.shields.io/badge/Leave%20Feedback-555555?style=flat)](https://app.dosu.dev/response-feedback?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=knowledge-infrastructure-feedback&utm_term=opendatalab%2FMinerU&message_id=5a01aa5f-a398-413b-89e3-bf62680d8048) [![Learn about MinerU with Dosu](https://img.shields.io/badge/Learn%20about%20MinerU%20with%20Dosu-2f7b3f?style=flat&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CPHN2ZyB3aWR0aD0iODYiIGhlaWdodD0iODkiIHZpZXdCb3g9IjAgMCA4NiA4OSIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48cGF0aCBkPSJNNS4yOTIzNiAxMi43OTI4TDE3Ljc1OTMgNi42ODE4OFY3Mi41NjY3TDUuMjkyMzYgODQuMDYxOFYxMi43OTI4WiIgZmlsbD0iI0I0QkI5MSIvPjxwYXRoIGQ9Ik0xOC4yNTc1IDczLjExOTZMNTkuMTMyOSA3Mi43NDhMNTEuNzAxMSA4Mi40MDk1TDI5LjAzMzggODYuMjkxTDYuMjM5NjIgODUuMTU1NEwxOC4yNTc1IDczLjExOTZaIiBmaWxsPSIjNzc4NTYxIi8%2BPHBhdGggZD0iTTE3LjQ5MTYgMy43MzYzM0wzLjU4NTU3IDEyLjcwOTlWODMuNTc5MkMzLjU4NTU3IDg0Ljc1NDIgNC45ODU2MyA4NS4zNjUyIDUuODQ3MDUgODQuNTY2TDE5LjYyOTYgNzEuNzgwMSIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSI2LjQyODQ0IiBzdHJva2UtbGluZWNhcD0icm91bmQiLz48bWFzayBpZD0iZG9zdS1kLWN1dG91dCIgZmlsbD0id2hpdGUiPjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNNDAuNzA0IDAuNTE4MDY2SDE3LjA0MzlWNzYuMjIyMUg0MC43MDRINDIuNTgwNUg0Ny44MDEzQzY4LjcwNjQgNzYuMjIyMSA4NS42NTMzIDU5LjI3NTIgODUuNjUzMyAzOC4zNzAxQzg1LjY1MzMgMTcuNDY1IDY4LjcwNjMgMC41MTgwNjYgNDcuODAxMyAwLjUxODA2Nkg0Mi41ODA1SDQwLjcwNFoiLz48L21hc2s%2BPHBhdGggZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik00MC43MDQgMC41MTgwNjZIMTcuMDQzOVY3Ni4yMjIxSDQwLjcwNEg0Mi41ODA1SDQ3LjgwMTNDNjguNzA2NCA3Ni4yMjIxIDg1LjY1MzMgNTkuMjc1MiA4NS42NTMzIDM4LjM3MDFDODUuNjUzMyAxNy40NjUgNjguNzA2MyAwLjUxODA2NiA0Ny44MDEzIDAuNTE4MDY2SDQyLjU4MDVINDAuNzA0WiIgZmlsbD0iI0YzRjZGMSIvPjxwYXRoIGQ9Ik0xNy4wNDM5IDAuNTE4MDY2Vi02LjU3OTE5SDkuOTQ2NjlWMC41MTgwNjZIMTcuMDQzOVpNMTcuMDQzOSA3Ni4yMjIxSDkuOTQ2NjlWODMuMzE5NEgxNy4wNDM5Vjc2LjIyMjFaTTE3LjA0MzkgNy42MTUzMkg0MC43MDRWLTYuNTc5MTlIMTcuMDQzOVY3LjYxNTMyWk0yNC4xNDEyIDc2LjIyMjFWMC41MTgwNjZIOS45NDY2OVY3Ni4yMjIxSDI0LjE0MTJaTTQwLjcwNCA2OS4xMjQ5SDE3LjA0MzlWODMuMzE5NEg0MC43MDRWNjkuMTI0OVpNNDIuNTgwNSA2OS4xMjQ5SDQwLjcwNFY4My4zMTk0SDQyLjU4MDVWNjkuMTI0OVpNNDcuODAxMyA2OS4xMjQ5SDQyLjU4MDVWODMuMzE5NEg0Ny44MDEzVjY5LjEyNDlaTTc4LjU1NiAzOC4zNzAxQzc4LjU1NiA1NS4zNTU1IDY0Ljc4NjcgNjkuMTI0OSA0Ny44MDEzIDY5LjEyNDlWODMuMzE5NEM3Mi42MjYxIDgzLjMxOTQgOTIuNzUwNSA2My4xOTQ5IDkyLjc1MDUgMzguMzcwMUg3OC41NTZaTTQ3LjgwMTMgNy42MTUzMkM2NC43ODY2IDcuNjE1MzIgNzguNTU2IDIxLjM4NDcgNzguNTU2IDM4LjM3MDFIOTIuNzUwNUM5Mi43NTA1IDEzLjU0NTMgNzIuNjI2IC02LjU3OTE5IDQ3LjgwMTMgLTYuNTc5MTlWNy42MTUzMlpNNDIuNTgwNSA3LjYxNTMySDQ3LjgwMTNWLTYuNTc5MTlINDIuNTgwNVY3LjYxNTMyWk00MC43MDQgNy42MTUzMkg0Mi41ODA1Vi02LjU3OTE5SDQwLjcwNFY3LjYxNTMyWiIgZmlsbD0iYmxhY2siIG1hc2s9InVybCgjZG9zdS1kLWN1dG91dCkiLz48cGF0aCBkPSJNNjguOTIxNSAzNi4wMTM1QzY4LjkyMTUgMzYuMDEzNSA2NS43MzY5IDQ5LjQ3MzggNTEuNDIzMSA0OS40NzM4QzM3LjEwOTMgNDkuNDczOCAzMi41Nzg3IDM3LjM1OTYgMzIuNTc4NyAzNi4wMTM1IiBzdHJva2U9ImJsYWNrIiBzdHJva2Utd2lkdGg9IjcuNjkxNjEiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCIvPjxwYXRoIGQ9Ik0wLjM0ODYzMyA4NS40OTQ2QzAuMzQ4NjMzIDg1LjQ5NDYgMjkuNDg1NiA4NS44MzA5IDM0LjgwOSA4NS42OThDNDQuODMzNyA4NS40NDc3IDUxLjI4NzIgODQuNDAyIDU3LjUyNjkgNzguOTcyNEM2Mi44MTI5IDc0LjM3MjcgNzUuMTM0MiA1OS42ODM2IDc1LjEzNDIgNTkuNjgzNiIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSI2LjE2NDgyIi8%2BPC9zdmc%2B)](https://github.dosu.com/opendatalab/MinerU?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=knowledge-infrastructure-learn-repo&utm_term=opendatalab%2FMinerU) [![Add Dosu to your team](https://img.shields.io/badge/Add%20Dosu%20to%20your%20team-1f6feb?style=flat)](https://app.dosu.dev/signup?referrer=openSource&source=github-footer&utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=knowledge-infrastructure-add-team&utm_term=opendatalab%2FMinerU)

**Reply by LeeLear** [reply]
root@5e15ef7a7bd7:/usr/local/python3.11.13/bin# mineru-vllm-server --port 30000 --method ocr
INFO 01-08 09:21:09 [__init__.py:43] Available plugins for group vllm.platform_plugins:
INFO 01-08 09:21:09 [__init__.py:45] - ascend -> vllm_ascend:register
INFO 01-08 09:21:09 [__init__.py:48] All plugins in this group will be loaded. Set `VLLM_PLUGINS` to control which plugins to load.
INFO 01-08 09:21:09 [__init__.py:217] Platform plugin ascend is activated
INFO 01-08 09:21:09 [importing.py:68] Triton not installed or not compatible; certain GPU-related functions will not be available.
2026-01-08 09:21:16.741 | INFO     | mineru.backend.vlm.utils:enable_custom_logits_processors:46 - compute_capability: 8.0 >= 8.0 and vllm version: 0.13.0 >= 0.10.1, enable custom_logits_processors
start vllm server: ['/usr/local/python3.11.13/bin/mineru-vllm-server', 'serve', '/root/.cache/modelscope/hub/models/OpenDataLab/MinerU2___5-2509-1___2B', '--port', '30000', '--method', 'ocr', '--gpu-memory-utilization', '0.5', '--logits-processors', 'mineru_vl_utils:MinerULogitsProcessor']
INFO 01-08 09:21:18 [__init__.py:108] Registered model loader `<class 'vllm_ascend.model_loader.netloader.netloader.ModelNetLoaderElastic'>` with load format `netloader`
usage: mineru-vllm-server [-h] [-v] {chat,complete,serve,bench,collect-env,run-batch} ...
mineru-vllm-server: error: unrecognized arguments: --method ocr

**Reply by myhloli** [reply]
vlm模式不支持这个参数，这个bot乱编的，直接按官方教程操作下呢

**Comment by myhloli** [comment]
按官方教程试一下呢https://opendatalab.github.io/MinerU/zh/usage/acceleration_cards/Ascend/
